docs/glossary.rst

```diff
@@ -306,18 +306,36 @@ Glossary
      a principal, but this is not strictly necessary in custom policies that
      define their principals differently.
 
+   identity
+     An identity is an opaque identifier of the user associated with the
+     current request.
+
+   security policy
+     A security policy in :app:`Pyramid` terms is a bit of code which has an
+     API which identifies the user associated with the current request (perhaps
+     via a cookie or ``Authorization`` header) and determines whether or not
+     that user is permitted to access the requested resource.
+
    authorization policy
      An authorization policy in :app:`Pyramid` terms is a bit of
      code which has an API which determines whether or not the
      principals associated with the request can perform an action
      associated with a permission, based on the information found on the
      :term:`context` resource.
 
+     .. deprecated:: 2.0
+       Authorization policies have been deprecated in favor of a
+       :term:`security policy`.
+
    authentication policy
      An authentication policy in :app:`Pyramid` terms is a bit of
      code which has an API which determines the current
      :term:`principal` (or principals) associated with a request.
 
+     .. deprecated:: 2.0
+       Authentication policies have been deprecated in favor of a
+       :term:`security policy`.
+
```

Comment:
I don't want to add stop energy here but I will :)  I am a big -1 on gluing together authentication and authorization into a single policy object.  Thoughts?

Reply:
I think you need to expand on why you are concerned about this. A -1 with no reasoning isn’t very helpful. There is a lot of discussion in https://github.com/Pylons/pyramid/issues/3422#issuecomment-457861409 about why combining them provides a lot of benefits and discusses issues people are running into where 1) the only way to pass data between authn and authz policies right now is via principal strings as well as 2) subclassing of authn policies.

Reply:
I've been thinking about this change a lot, and the positives and improvements that this change brings outweigh the negative effects of gluing the two together.

For many of my own projects these changes will provide greater flexibility to better match the way permissions work in my app that are not strictly ACL and/or based upon principals.